dolphinscheduler-api-test/dolphinscheduler-api-test-case/src/test/java/org/apache/dolphinscheduler/api.test/cases/WorkFlowShellAPITest.java

```diff
@@ -0,0 +1,131 @@
+
+@DolphinScheduler(composeFiles = "docker/basic/docker-compose.yaml")
+@Slf4j
+public class WorkFlowShellAPITest {
+    private static final Logger logger = LoggerFactory.getLogger(ProjectAPITest.class);
+
+    private static final String projectName = "case02_wen";
+
+    private static final String projectDesc = "123";
+
+    private static final String workFlowName = "shell123";
+
+    private static final String tenantName = "wendada";
+
+    private static final String user = "admin";
+
+    private static final String password = "dolphinscheduler123";
+
+    private static String sessionId = null;
+
+    @BeforeAll
+    public static void setup() {
+        LoginPage loginPage = new LoginPage();
+        ProjectPage projectPage = new ProjectPage();
+        HttpResponse loginHttpResponse = loginPage.login(user, password);
+        TenantPage tenantPage = new TenantPage();
+
+        sessionId = JSONUtils.convertValue(loginHttpResponse.body().data(), LoginResponseData.class).sessionId();
+        projectPage.createProject(sessionId, projectName, projectDesc, user);
+        tenantPage.createTenant(sessionId, tenantName, 1, "");
+
+    }
+
+    @Test
+    @Order(1)
+    public void testCreateWorkflow() {
+        WorkFlowDefinitionPage flow = new WorkFlowDefinitionPage();
+        flow.getGenNumId(sessionId, projectName);
+        HttpResponse res = flow.createWorkflow(sessionId, projectName, workFlowName);
+
+        logger.info("Create workflow res：%s", res);
+        Assertions.assertTrue(res.body().success());
+    }
+
+    @Test
+    @Order(2)
+    public void testOnlineWorkflow() {
+        WorkFlowDefinitionPage flow = new WorkFlowDefinitionPage();
+        HttpResponse res = flow.onLineWorkflow(sessionId, projectName, workFlowName);
+
+        logger.info("Online workflow res：%s", res);
+        Assertions.assertTrue(res.body().success());
+
+    }
+
+    @Test
+    @Order(3)
+    public void testQueryWorkflow() {
+        WorkFlowDefinitionPage flow = new WorkFlowDefinitionPage();
+        flow.getGenNumId(sessionId, projectName);
+        HttpResponse res = flow.queryWorkflow(sessionId, projectName, workFlowName);
+
+        logger.info("Query workflow res：%s", res);
+        Assertions.assertTrue(res.body().success());
+    }
+
+    @Test
+    @Order(4)
+    public void testRunWorkflow() {
+        WorkFlowDefinitionPage flow = new WorkFlowDefinitionPage();
+        HttpResponse res = flow.runWorkflow(sessionId, projectName, workFlowName);
+
+        logger.info("Run workflow res：%s", res);
+        Assertions.assertTrue(res.body().success());
+
+    }
+
+    @Test
+    @Order(5)
+    public void testQueryWorkflowInstanceState() throws InterruptedException {
+        WorkFlowInstancesPage instance = new WorkFlowInstancesPage();
+        String state = null;
+
+        for (int i=0; i<5; i++){
+            state = instance.queryWorkflowInstanceState(sessionId, projectName, workFlowName);
+            Thread.sleep(1000);
+        }
```

Comment:
The waiting time and cycle times of querying workflow instances may be used in multiple tests. So I think we should put this into constants for public use.